WeakAuras/Prototypes.lua

```diff
@@ -10702,6 +10701,183 @@ Private.event_prototypes = {
+      end
+      return {
+        ["events"] = events,
+      }
+    end,
+    internal_events = function(trigger, untrigger)
+      local events = {"INSTANCE_LOCATION_CHECK"};
+      if trigger.use_instance_difficulty ~= nil
+         or trigger.use_instance_type ~= nil
+         or trigger.use_instance_size ~= nil
+      then
+        tinsert(events, "WA_DELAYED_PLAYER_ENTERING_WORLD")
+      end
+      return events;
+    end,
+    force_events = "INSTANCE_LOCATION_CHECK",
+    name = WeakAuras.newFeatureString..L["Location"],
+    init = function(trigger)
+      local ret = [=[
+        local recursive = %s
+        local zoneChecker = Private.ExecEnv.ParseZoneCheck(%q, recursive)
+
+        local uiMapId = C_Map.GetBestMapForUnit("player")
+        local zonegroupId = uiMapId and C_Map.GetMapGroupID(uiMapId)
+        local instanceName, _, _, _, _, _, _, instanceId = GetInstanceInfo()
+        local minimapZoneText = GetMinimapZoneText()
+
+        local zoneText = GetZoneText()
+        local mapInfo = uiMapId and C_Map.GetMapInfo(uiMapId)
+        local isIndoors = IsIndoors()
+      ]=]
+      return ret:format(trigger.use_recursive and "true" or "nil", trigger.zoneIds or 0)
+    end,
+    statesParameter = "one",
+    args = {
+      {
+        name = "zoneIds",
+        display = L["Player Location ID(s)"],
+        type = "string",
+        multiline = true,
+        desc = get_zoneId_list,
+        test = "zoneChecker:Check(uiMapId, zonegroupId, instanceId, minimapZoneText)",
+      },
+      {
+        name = "recursive",
+        display = L["Recursive"],
+        desc = L["Enable this to recurse on each child map until it finds a match. Disable this to only include direct descendants."],
+        type = "toggle",
+        test = "true",
+        enable = function(trigger)
+          if trigger.use_zoneIds then
+            local zoneChecker = Private.ExecEnv.ParseZoneCheck(trigger.zoneIds)
+            return zoneChecker and zoneChecker.isChildFilter
+          end
+        end,
+      },
+      {
+        name = "zoneId",
+        display = L["Zone ID"],
+        conditionType = "number",
+        init = "uiMapId",
+        store = true,
+        hidden = true,
+        test = "true",
+        noProgressSource = true,
+      },
+      {
+        name = "zone",
+        display = L["Zone Name"],
+        type = "string",
+        conditionType = "string",
+        store = true,
+        init = "zoneText",
+        multiEntry = {
+          operator = "or",
+        }
+      },
+      {
+        name = "subzone",
+        display = L["Subzone Name"],
+        desc = L["Name of the (sub-)zone currently shown above the minimap."],
+        type = "string",
+        conditionType = "string",
+        store = true,
+        init = "minimapZoneText",
+        multiEntry = {
+          operator = "or",
+        },
+      },
+      {
+        name = "mapType",
+        display = L["Map Type"],
```

Comment:
Do you have an use case in mind? Or did you add it just for completeness? To me it feels a bit too niche.